### ISSUE TITLE
Completely silent mode

### ISSUE THREAD

**Issue by josh-braid** [nrwl/nx#14524]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description

Currently, if I implement some script target using `nx:run-commands`, depending on a `build` target then no matter what, when I `nx run` it, it will include a certain amount of spam about the build process, caching etc.

It would be very nice to be able to have a `--silent` option for script targets that need to control their output. (Ideally on the executor so it doesnt need to be passed every time it's run)

## Motivation

I'd like to avoid needing wrapper scripts that do `nx build thing && node dist/{thingPath}` since encoding that dependency is redundant and seems like it's exactly the thing you want a build system to do.

## Suggested Implementation

Not sure about internally implementing it, seems like silencing output is straightforward. In terms of the interface, it would be nice to have both an `nx run --silent <target>` and also an option for `"silent": true` in the executor configuration.

## Alternate Implementations

I can keep wrapping nx calls with bash scripts, though that makes me kind of sad :(

**Comment by josh-braid** [comment]
Any update on whether this is planned?

**Comment by ryanouellette** [comment]
+1. I would also find this immensely useful for essentially the same reasons. It would be really nice to utilize `nx run` for building, if necessary, CLI tools and consuming their output without NX related noise going to stdout/stderr.

+1 to also being able to specify an option in the executor configuration to suppress the output by default for a target.

**Comment by qingquanwang** [comment]
+1

**Comment by lppedd** [comment]
Bumping it for visibility. Great feature imo.

**Comment by martin-mindflow** [comment]
This would be extremely useful for our CLI tools, and we'd be able to remove the little hacks we had to resort to to get plain outputs for commands (for example when we need to pipe one command to another)

**Comment by antoineviscardi** [comment]
As a workaround, I believe the following works

Instead of this,
`nx run some-project:wrapped-script --arg myarg > out.txt` 

This,
`nx run some-project:wrapped-script "--arg myarg > out.txt"`

I think this is still far from ideal and quite brittle, but if your piping is simple, it seems good enough.

**Comment by Tramb-dev** [comment]
I agree with you; it's unfortunate that we can't directly retrieve console output without all the additional data provided by Nx. I currently need this feature in my executor to fetch a raw string used by my CI. It would be beneficial to implement a functionality similar to the 'nx show projects' command, which outputs a string without the command line information that was executed.

**Comment by cxsimon** [comment]
+1 on this, the output of nx is spamming the cli controllers to a point where its impossible to load the pages via Jenkins

**Comment by hco** [comment]
I also would love this feature and thought about it a bit, too.

> Also, it'd be good if there is an error in the build that output does get printed, but only on error.

I think this should be a bit configurable, e.g. which exit codes mean "error" and maybe also "if a specific string shows up in the output, consider it an error"

This would need to be implemented in rust right now, right?

**Comment by ClementMindflow** [comment]
Same here, invoking a CLI from nx is super powerful as it benefit out of the box of env var autoload and task dependency execution.
However the amount of logs coming from nx makes this approach nearly un usable for large CLI with a lots of deps.

**Comment by FrozenPandaz** [comment]
We could look into implementing this as `--outputStyle none`.

**Comment by hco** [comment]
@FrozenPandaz Would that work on a per-target basis? Looking at the original ticket, the author probably has two targets: `serve` and `build`, `serve` depends on `build` and they only want to hide the output of `build`

**Comment by mrfelton** [comment]
100% on this. Kinda surprised this one has not been addressed yet. Presumably every single person that needs to do anything remotely complex with nx now has their own custom wrapper scripts to cut out all the nx noise and just get the output of a given command.

Here is a concrete example of a simple nx target that outputs some json, and the custom crap we have to wrap an `nx run` call with every time we run it.

Default (unusable) output:
```
❯ nx packages

> nx run zapmiddleware.lightning.api:packages


^D
["zaplightning"]

—————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Successfully ran target packages for project zapmiddleware.lightning.api (2s)
```

Wrapped:
```
❯ nx packages | grep -o '\[.*\]'
["zaplightning"]
```

**Comment by mrfelton** [comment]
Cross linking a couple of similar requests/discussion for reference

- https://github.com/nrwl/nx/issues/1571
- https://github.com/nrwl/nx/discussions/15198

**Comment by johnathanstewartmlb** [comment]
+1

**Comment by stefan-schweiger** [comment]
This is still a issue as it makes certain scripting / CI tasks unnecessary hard

**Comment by hyperb1iss** [comment]
Bumping in June 2025, it would be stellar to have this

**Comment by lukeramsden** [comment]
Still can't make nx shut up. Why are you building a pointless `tui` mode instead of letting nx get out of my way and shut up?

**Comment by adamjosefus** [comment]
Same :/

**Comment by lukeramsden** [comment]
`nx` is incredibly poor for being used by AI agents because the output is such low signal

**Comment by lukeramsden** [comment]
This is what I want it to do: https://gist.github.com/lukeramsden/db8e76637b7f6e188e00c45b113e97d4

**Comment by stefan-schweiger** [comment]
Still an issue

**Comment by ValentinGurkov** [comment]
Still an issue. Our microservices monorepo is quite large, the GitHub CI can barely load the logs with this much spam and we've opted for writing the logs in the file and parsing the logs programatically to print what is relevant instead...

**Comment by lukeramsden** [comment]
nx is not a serious tool lol very sad

**Comment by AgentEnder** [comment]
We are working on something in this area, its not exactly a completely silent mode but is along those lines, specifically for working with AI agents context windows better. It is not really what this ticket was initially for, in that it was asking for a way to suppress the output of dependent tasks rather than what the conversation has evolved into.

The new output will idealistically look something like the checklist / summary view that you get at the end or run-many / affected's dynamic output styles today, but with no task logs at all (including for failed tasks), and instead showing the filepath to that task's logs as stored on disk in the cache store. 

The original request of supporting silencing the wrapper bits that Nx adds to terminal output isn't something we've decided on a direction for. There appears to be a few competing requests that are being represented here:
 
- Hide dependent tasks output and all decoration added by Nx, while keeping the output of the specified task in `nx run`
- Hide all decoration, but not dependent tasks output... potentially applicable in run, run-many, and affected
- Reduce context rot in AI agents by avoiding pollution from printing every task's output in full

That third one is what we are actively working on.

**Comment by lukeramsden** [comment]
That's a good start but the first and second one are also important
